Basic-Car-Maintenance/Shared/Onboarding/Views/WelcomeViewAddVehicle.swift

```diff
@@ -9,99 +9,177 @@ import SwiftUI
             
-            Spacer(minLength: 10)
+            List {
+                vehicleDetailRow(title: "Name", text: $vehicleName, alert: $validationAlertName)
+                vehicleDetailRow(title: "Make", text: $vehicleMake, alert: $validationAlertMake)
+                vehicleDetailRow(title: "Model", text: $vehicleModel, alert: $validationAlertModel)
+            }
+            .frame(maxHeight: 200)
+            .scrollContentBackground(.hidden)
+        }
+        .padding(.horizontal, 15)
+    }
+    
+    private func vehicleDetailRow(title: String, text: Binding<String>, alert: Binding<Bool>) -> some View {
+        LabeledContent {
+            TextField("Vehicle \(title)", text: text)
+                .alert(
+                    "Vehicle \(title) cannot be empty! 🚗",
+                    isPresented: alert
+                ) {
+                    Button("OK", role: .cancel) {}
+                }
+                .frame(width: 200)
+        } label: {
+            HStack {
+                Text(title)
+                Spacer()
+            }
+        }
+        .showClearButton(text)
+    }
+    
+    private var bottomText: some View {
+        Text("You can edit more data about the vehicle in the 'Settings' tab.")
+            .foregroundStyle(.gray)
+            .padding(.horizontal, 25)
+    }
+    
+    private var addVehicleButton: some View {
+        Button {
+            addVehicle()
+        } label: {
+            Text("Add Vehicle")
+                .fontWeight(.bold)
+                .foregroundStyle(.white)
+                .frame(maxWidth: .infinity)
+                .padding(.vertical, 14)
+                .background(Color("basicGreen").gradient, in: .rect(cornerRadius: 12))
+                .contentShape(.rect)
+        }
+        .padding(15)
+        .padding(.horizontal, 15)
+        .alert(
+            Text("Congratulations! 🎉 🚙"),
+            isPresented: $onboardingViewCompletedAlert
+        ) {
+            Button("OK", role: .cancel, action: {
+                isFirstTime = false
+            })
+        } message: {
+            Text(
+                "From now on, next screen will be the main app screen.",
+                comment: "Shows popup when completed Onboarding tutorial"
+            )
+        }
+    }
+    
+    func addVehicle() {
+        if vehicleName.isEmpty {
+            validationAlertName = true
+        } else if vehicleMake.isEmpty {
+            validationAlertMake = true
+        } else if vehicleModel.isEmpty {
+            validationAlertModel = true
+        } else {
+            UserDefaults.standard.set(vehicleName, forKey: "vehicleName")
+            UserDefaults.standard.set(vehicleMake, forKey: "vehicleMake")
+            UserDefaults.standard.set(vehicleModel, forKey: "vehicleModel")
             
+//            isFirstTime = false
+            onboardingViewCompletedAlert = true
+        }
+    }
+    
+    private var backButton: some ToolbarContent {
```

Comment:
don't use a custom back button, us the default
This is because it doesn't match other defaults in the app and then this one looks like the odd one out

Reply:
Ok, I will do it 👍